dxr/build.py

```diff
@@ -507,40 +506,50 @@ def index_file(tree, tree_indexers, path, es, index, jinja_env):
             es.bulk_index(index, LINE, chunk_of_needles, id_field=None)
 
     # Render some HTML:
-    # TODO: Make this no longer conditional on is_text, and come up with a nice
-    # way to show binary files, especially images.
-    if is_text and 'html' not in tree.config.skip_stages:
-        _fill_and_write_template(
-            jinja_env,
-            'file.html',
-            join(tree.target_folder, rel_path + '.html'),
-            {# Common template variables:
-             'wwwroot': tree.config.wwwroot,
-             'tree': tree.name,
-             'tree_tuples': [(t.name,
-                              parallel_url(t.name, tree.config.wwwroot, rel_path),
-                              t.description)
-                             for t in tree.config.sorted_tree_order],
-             'generated_date': tree.config.generated_date,
-             'google_analytics_key': tree.config.google_analytics_key,
-             'filters': filter_menu_items(),
-
-             # File template variables:
-             'paths_and_names': linked_pathname(rel_path, tree.name),
-             'icon': icon(rel_path),
-             'path': rel_path,
-             'name': os.path.basename(rel_path),
-
-             # Someday, it would be great to stream this and not concretize the
-             # whole thing in RAM. The template will have to quit looping
-             # through the whole thing 3 times.
-             'lines': zip(build_lines(contents, refs, regions),
-                          annotations_by_line) if is_text else [],
-
-             'is_text': is_text,
-
-             'sections': build_sections(links)})
+    if 'html' not in tree.config.skip_stages:
+        # Common template variables:
+        common = {
+            'wwwroot': tree.config.wwwroot,
+            'tree': tree.name,
+            'tree_tuples': [(t.name,
+                parallel_url(t.name, tree.config.wwwroot, rel_path), t.description)
+                for t in tree.config.sorted_tree_order],
+            'generated_date': tree.config.generated_date,
+            'google_analytics_key': tree.config.google_analytics_key,
+            'filters': filter_menu_items(),
+        }
+
+        # File template variables
+        file_vars = {
+            'paths_and_names': linked_pathname(rel_path, tree.name),
+            'icon': icon(rel_path),
+            'path': rel_path,
+            'name': os.path.basename(rel_path),
+        }
+
+        if is_image(rel_path):
+            _fill_and_write_template(
+                jinja_env,
+                'image_file.html',
+                join(tree.target_folder, rel_path + '.html'),
+                merge(common, file_vars, {
+                'image_src': raw_url(tree.name, rel_path) }))
+
+        else:
```

Comment:
I'd been basing this off of [bug 1052217](https://bugzilla.mozilla.org/show_bug.cgi?id=1052217), where you mentioned it would be not bad to acknowledge existence of other binary files, so I removed the is_text check here. The template already has is_text check and shows (binary file) if the file is not text. Actually there is a test for this at [test_binary_files.py](http://git.io/mJA1BA).